### DISCUSSION TITLE
Ideas for example apps

### DISCUSSION THREAD

**Discussion by jchris** [fireproof-storage/fireproof#6] [Ideas]
I am doing the next round of example apps, here are some ideas. The hope is that they can act as starter kits, and inspire people to start with them and innovate in their own ways.

### FireSign - a Fireproof powered DocuSign alternative
**FireSign** revolutionizes the way documents are signed and stored. Leveraging the power of Fireproof, it allows users to upload PDFs, mark signature locations, and obtain a cryptographically verifiable archive. With an emphasis on security and convenience, FireSign is not just an alternative to existing platforms but a beacon for digitally transforming workflows in various industries.

### ✅ Upload gallery - where you can upload as many images or videos as you want, and then browse them
Embrace limitless creativity with the **Upload gallery**, a robust platform where users can upload and browse an infinite collection of images and videos. Powered by Fireproof's secure and scalable technology, it's more than just a gallery; it's a space for artists, photographers, and multimedia enthusiasts to explore, share, and collaborate without any bounds. [Try it now at Public Image Gallery](https://public-media.fireproof.storage/) -- once we remove the file size limit for encrypted files we can do a private one.

### FireScript - A Nodebook for JavaScript with Instant Replay
**FireScript** is a transformative development environment inspired by Jupyter Notebooks but crafted exclusively for JavaScript. Utilizing Fireproof's unique capabilities, FireScript keys nodebook JS code off of the hash content address, enabling an instant replay feature. If you've executed a version of code before, FireScript has the data snapshot readily available. It's a platform designed for interactive data exploration, where the past and present of code execution coexist seamlessly.

### Real talk, a multimedia Twitter clone with public and private accounts
**Real talk** is the social media platform for the new era. Equipped with multimedia capabilities and options for both public and private accounts, it fosters genuine connections and conversations. Built on Fireproof's strong foundation, it offers reliability and security, inviting users to engage in real talk in a virtual world, free from the traditional barriers of social networking.

### FieldNotes - a basic form-filling app
Say goodbye to cumbersome paperwork with **FieldNotes**, the form-filling app that's anything but basic. Whether it's for business surveys, customer feedback, or administrative needs, FieldNotes simplifies the process by offering a seamless digital interface. Utilizing Fireproof's strengths, it ensures that data integrity is maintained, making it a vital tool for any modern organization.

### TodoMVC with multi-list and shared lists
Organize your life and collaborate effortlessly with **TodoMVC**. Offering multi-list functionality and shared lists, this application transforms the way tasks are managed. Whether it's a complex project or daily chores, TodoMVC, powered by Fireproof, provides a flexible and secure environment to plan, track, and achieve goals, enabling users to stay ahead of the curve in both personal and professional spheres.

### Smart Book - A Dynamic, Interactive Knowledge Base with AI-Powered Assistance
**Smart Book** is not just a repository of information but an engaging, intelligent companion in your learning journey. Beyond offering existing content, it dynamically interacts with users to identify gaps in understanding or information. If a query leads to a novel insight or uncovers a missing piece of knowledge, the Smart Book's AI-powered chat initiates the creation of new topics or notebooks. These new additions, crafted through interactive dialogue, make the Smart Book a continuously evolving resource. Powered by Fireproof, it ensures that the process is seamless, secure, and aligned with the users' needs. It's more than a knowledge base; it's an adaptive learning environment that grows and learns with you, fostering a culture of curiosity, exploration, and shared wisdom.

## The goal

Here are some examples of how people might rebuild these in the real world. The goal being to make it easy for people to quick start real apps. In building these examples, the Fireproof community will necessarily confront the challenges real app devs will find, and come up with solutions and enhancements for each use case.

### FireSign - a Fireproof powered DocuSign alternative
1. **LegalProof**: Specifically designed for legal firms, allowing for added layers of verification and compliance checks.
2. **EduSign**: For educational institutions to securely sign and store student documents, transcripts, and agreements.
3. **MediSign**: Targeted for the healthcare industry, providing secure, verifiable signing and storage for patient consent forms and medical documents.

### Upload gallery - where you can upload as many images or videos as you want, and then browse them
1. **ArtSpace**: A platform for artists to upload and showcase their portfolio, with features for categorizing and tagging different art forms.
2. **EventSnap**: Designed for event photographers, allowing them to upload event photos and videos and share private galleries with clients.
3. **FamilyHub**: A private gallery space for families to share and keep memories, including images, videos, and annotations.

### FireScript - An Interactive Nodebook for JavaScript, Powering Instant Replay and Version Control with Fireproof Snapshots
1. **Data Explorer with Historical Insights**: For researchers and analysts to effortlessly query and analyze data, with the ability to replay previous versions for deeper exploration and understanding.
2. **Real-Time Collaboration Studio**: A unique environment for development teams, allowing real-time collaboration with instant version control, enabling a living history of the project.
3. **Educational Suite with Replayable Lessons**: A dynamic learning platform designed for educators and students, transforming JavaScript learning into an engaging, hands-on experience with replayable lessons and interactive guidance.

### Real talk, a multimedia Twitter clone with public and private accounts
1. **TradeTalk**: A network specifically for traders and investors to share insights, multimedia analyses, and real-time market updates.
2. **ParentConnect**: A social platform for parents to share experiences, tips, and media about parenting in both public and private groups.
3. **FitnessFeed**: For fitness enthusiasts to share workout videos, nutrition tips, and personal progress.

### FieldNotes - a basic form-filling app
1. **SurveyMaster**: A tool for businesses to create and share various types of customer surveys, collecting and analyzing the data.
2. **TenantForms**: Aimed at landlords and property managers for streamlined tenant application and agreement processes.
3. **HealthCheck**: For clinics and hospitals to manage patient intake forms, appointments, and medical history records.

### TodoMVC with multi list and shared lists
1. **ProjectPulse**: Extending the todo-list into a project management tool with shared task lists, deadlines, and collaboration features.
2. **WeddingPlanner**: A specialized list-management app for wedding planning, allowing collaboration among family and vendors.
3. **StudyBuddy**: An app for students to manage and share study schedules, assignments, and collaborative group projects.

### Smart Book - A Dynamic, Interactive Knowledge Base with AI-Powered Assistance
1. **Screenwriting Team Companion**: Transforms traditional writing collaboration, providing an interactive platform for character development, plot crafting, and real-time brainstorming. A content-augmented AI guides writers through existing material, fostering creativity and cohesiveness.
2. **Company Handbook Reimagined**: Revolutionizes the corporate handbook by adding interactive elements like searchable policies, team structures, and project overviews. Employees can effortlessly access information, and the dynamic nature keeps the content evergreen and engaging.
3. **Educational Suite for Teachers and Students**: A comprehensive solution for education, enabling interactive lesson modules, assignments, and student discussions. The AI-supported chat acts as a virtual teaching assistant, providing explanations and guidance, enhancing the learning experience.

These derivative apps can further extend the reach and functionality of the original applications, catering to specific industries or needs, and leveraging the strengths of Fireproof's "build first, connect to the cloud when ready" concept.

**Comment by jchris** [comment]
Some notes in case anyone wants to do FireSign:

- Signatures https://github.com/szimek/signature_pad
- Overlay https://blog.logrocket.com/managing-pdfs-node-pdf-lib/#how-to-modify-existing-pdf
- To canvas https://projects.wojtekmaj.pl/react-pdf/

**Comment by jchris** [comment]
Just a chess game would be cool, and easy to upgrade for other uses.

**Comment by jchris** [comment]
Snapchat face filter library / chat / publishing

A snapchat style face filter using webcam (apparently easy https://codepen.io/livecodestream/pen/rNxrMzp ) and then let you customize and save filters as fireproof documetns. Then you can take snapshots and publish them pubically like in the gallery demo. This is a good demo because it is fast and local and still data you want to have privately sync and publically publishable

**Comment by drunkod** [comment]
@jchris ,  First and foremost. I think it would be better to add more examples of application authentication and authorization, with a focus on standarts web APIs... (passwordless, toch...)
 All that's holding the P2P industry back right now is a lack of best common practices.
 
The sample apps(frontend) you suggest are very much in open source examples on github.

**Reply by jchris** [reply]
@drunkod thanks for the comment 💯

My closest example for that right now is via the IPSF connector, which uses https://ucan.xyz for user-controlled self-sovereign auth. You can play with my Public Media example to see IPFS / web3.storage in action. (The role I am referring here to is that they validate email address and provide a capability delegator so that services can delegate to users via email addresses, but all that is in-principle, not centralized.) 

Try the public media example here and check out it's logged out screen for more background info: https://public-media.fireproof.storage

**Comment by jchris** [comment]
Magnetic poetry with PartyKit: https://github.com/partykit/sketch-magnetic-poetry
